File: test/test_cases/lambda3.lambda_counter.again.py
def build_dbl():
    n = 1
    return lambda: ("WARNING: the translation might be inaccurate", (n := n * 2, n)[-1])[-1]
dbl1 = build_dbl()
dbl2 = build_dbl()
print(dbl1())
print(dbl2())
print(dbl1())

File: test/test_cases/lambda3.lambda_counter.again.js
function buildDbl() {
  let n = 1;
  return () => {
    n = n * 2;
    return n;
  };
}
let dbl1 = buildDbl();
let dbl2 = buildDbl();
console.log(dbl1());
console.log(dbl2());
console.log(dbl1());